hip-service/Link/Patient/PatientRepository.cs

```diff
@@ -0,0 +1,58 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using hip_service.Discovery.Patient.Helpers;
+using hip_service.Discovery.Patient.models;
+
+namespace hip_service.Link.Patient
+{
+    public class PatientRepository
+    {
+        private readonly string _filePath;
+
+        public PatientRepository(string filePath)
+        {
+            _filePath = filePath;
+        }
+
+        
+        IEnumerable<Discovery.Patient.Model.Patient> GetAllPatientFromJson()
+        {
+            var patientsInfo = FileReader.ReadJson(_filePath);
+            return patientsInfo;
+        }
+
+        public Discovery.Patient.Model.Patient? GetPatientInfoWithReferenceNumber(string referenceNumber)
+        {
+            try
+            {
+                var patientsInfo = GetAllPatientFromJson();
+                var patient = patientsInfo.First(patient => patient.Identifier == referenceNumber);
+                return patient;
+            }
+            catch (Exception e)
+            {
+                return null;
```

Comment:
This is wrong. Should return Option<Patient> https://github.com/nlkl/Optional. Don't return null ever.